src/main/java/com/epam/aidial/core/service/PermissionsFetcher.java

```diff
@@ -0,0 +1,22 @@
+package com.epam.aidial.core.service;
+
+import com.epam.aidial.core.ProxyContext;
+import com.epam.aidial.core.data.ResourceAccessType;
+import com.epam.aidial.core.security.AccessService;
+import com.epam.aidial.core.storage.ResourceDescription;
+
+import java.util.Map;
+import java.util.Set;
+
+@FunctionalInterface
+public interface PermissionsFetcher {
+    PermissionsFetcher EMPTY = resources -> Map.of();
+
+    Map<ResourceDescription, Set<ResourceAccessType>> fetch(Set<ResourceDescription> resources);
+
+    static PermissionsFetcher of(ProxyContext context, AccessService accessService) {
+        return Boolean.parseBoolean(context.getRequest().getParam("permissions", "false"))
```

Comment:
I think we need to keep all request parameters in controllers and provide a generic service, any service may later use this functionality

Reply:
Done.